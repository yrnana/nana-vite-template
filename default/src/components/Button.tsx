import { css } from '@emotion/react';

const styles = css`
  padding: 1rem 1.5rem;
  border-radius: 10px;
  border: 0;
  background: #000;
  color: #fff;
  font-size: 1.2rem;
  cursor: pointer;
`;

const Button: React.FC = ({ children }) => {
  return <button css={styles}>{children}</button>;
};

export default Button;
