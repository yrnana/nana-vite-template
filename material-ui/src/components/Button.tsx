import React from 'react';
import { Button as MuiButton } from '@material-ui/core';

const Button: React.FC = ({ children }) => {
  return (
    <MuiButton
      variant="contained"
      size="large"
      sx={{
        fontSize: 18,
      }}
    >
      {children}
    </MuiButton>
  );
};

export default Button;
