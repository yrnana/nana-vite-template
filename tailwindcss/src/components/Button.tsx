import React from 'react';

const Button: React.FC = ({ children }) => {
  return (
    <button className="py-2 px-4 font-semibold rounded-lg shadow-md text-white bg-green-500 hover:bg-green-700">
      {children}
    </button>
  );
};

export default Button;
