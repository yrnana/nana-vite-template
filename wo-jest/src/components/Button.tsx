import { css } from '@emotion/react';

const styles = css`
  padding: 1rem 1.5rem;
  font-size: 18px;
`;

type ButtonProps = {
  children?: React.ReactNode;
};

const Button = ({ children }: ButtonProps) => {
  return <button css={styles}>{children}</button>;
};

export default Button;
