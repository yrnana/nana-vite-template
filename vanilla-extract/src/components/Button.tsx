import * as styles from './Button.css';

const Button: React.FC = ({ children }) => {
  return <button className={styles.button}>{children}</button>;
};

export default Button;
