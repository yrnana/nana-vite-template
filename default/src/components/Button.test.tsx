import React from 'react';
import { render, screen } from '@testing-library/react';
import Button from './Button';

describe('버튼', () => {
  it('이전 스냅샷과 일치한다.', () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });

  it('버튼이 존재한다.', () => {
    render(<Button />);
    expect(screen.getByRole('button')).toBeInTheDocument();
  });
});
