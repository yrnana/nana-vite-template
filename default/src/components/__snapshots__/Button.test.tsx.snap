// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`버튼 이전 스냅샷과 일치한다. 1`] = `
.emotion-0 {
  padding: 1rem 1.5rem;
  border-radius: 10px;
  border: 0;
  background: #000;
  color: #fff;
  font-size: 1.2rem;
  cursor: pointer;
}

<div>
  <button
    class="emotion-0"
  />
</div>
`;
